scripts/test.sh

```diff
@@ -7,5 +7,9 @@ make -j8
 if [ -n "$BUILD_ONLY" ]; then
   echo "Skipping test run because of BUILD_ONLY environment variable"
 else
-  CTEST_OUTPUT_ON_FAILURE=1 make -j1 qa
+  if [ "$TRAVIS_BRANCH" = "master" ]; then
+    CTEST_OUTPUT_ON_FAILURE=1 CTEST_PARALLEL_LEVEL=4 make -j4 qa
+  else
+    CTEST_OUTPUT_ON_FAILURE=1 CTEST_PARALLEL_LEVEL=4 make -j4 check
+  fi
```

Comment:
Dumb question, but why do we need to run 'qa' on master and 'check' on other branches?